src/io/projection.hpp

```diff
@@ -106,7 +116,34 @@ auto ComputePlaneProjection(amrex::Vector<amrex::MultiFab> const &state_new, con
 	return proj_host;
 }
 
-void WriteProjection(const amrex::Direction dir, std::unordered_map<std::string, amrex::BaseFab<amrex::Real>> const &proj, amrex::Real time, int istep);
+void WriteProjection(amrex::Direction dir, std::unordered_map<std::string, amrex::BaseFab<amrex::Real>> const &proj, amrex::Real time, int istep,
+		     const std::string &basename, const YAML::Node &simulationMetadata);
+
+// Overload with particle support
+template <typename problem_t>
+void WriteProjection(amrex::Direction dir, std::unordered_map<std::string, amrex::BaseFab<amrex::Real>> const &proj, amrex::Real time, int istep,
+		     const std::string &basename, quokka::PhysicsParticleRegister<problem_t> &particleRegister, const std::vector<std::string> &particleTypes,
+		     const YAML::Node &simulationMetadata)
+{
+	BL_PROFILE("quokka::diagnostics::WriteProjection(with particles)");
+
+	// First, write the projection data using the base function (includes metadata)
+	WriteProjection(dir, proj, time, istep, basename, simulationMetadata);
+
+	// If no particle types specified, skip particle output
+	if (particleTypes.empty()) {
+		return;
+	}
+
+	// Construct the plotfile name (same directory as field data)
+	const std::string filename = amrex::Concatenate(basename, istep, 5);
+
+	// Write particles using the filtered method
+	constexpr int max_particles_to_write = 10000;
```

Comment:
If users have to turn on particle output for the projection file explicitly, then maybe limiting to the first 1e4 particles isn't necessary.